bytecomp/translmod.ml

```diff
@@ -656,8 +641,30 @@ and transl_structure loc fields cc rootpath final_env = function
                transl_module Tcoerce_none None modl, body),
           size
 
+      | Tstr_open od ->
+          begin match od.open_expr.mod_desc with
+          | Tmod_ident _ ->
+            transl_structure loc fields cc rootpath final_env rem
+          | _ ->
+            let ids = bound_value_identifiers od.open_type in
+            let mid = Ident.create_local "open" in
+            let rec rebind_idents pos newfields = function
+                [] ->
+                transl_structure loc newfields cc rootpath final_env rem
+              | id :: ids ->
+                let body, size =
+                  rebind_idents (pos + 1) (id :: newfields) ids
```

Comment:
I don't think these items should be added to `newfield`s -- I think that makes them part of the final module.

Reply:
Note that for the native backend, we do have to make them part of the final module, otherwise we would be breaking the native toplevel.
(Another option would be to pass a flag to know whether we're in the toplevel or not, and behave differently depending on that)

Reply:
Same as above -- they should actually be added to newfields.